### ISSUE TITLE
Default chunk_size and cores to ppj

### ISSUE THREAD

**Issue by pipitone** [CoBrALab/qbatch#79]
Back in issue #49 we discussed having chunk size defaulting to the ppj, but we lost track of that issue so I'm filing this. 

I'm thinking it the logic should be: unless overridden in the environment or command-line, `-c` and `-j` both default to `--ppj`. That way, you default to running as many commands in parallel in a job as you have processors allocated, and the user only needs to change one option, `--ppj`, if they want to scale that up or down.

**Comment by pipitone** [comment]
Er, to be clear, setting `--ppj` on the command line should adjust `-j` and `-c` unless they are also set. This isn't the case right now.

**Comment by gdevenyi** [comment]
Yes, definitely agree.

Sidenote: do we handle infinite chunk size yet? I think we said if chunks are "0" we allow arbitrarily large chunks?

**Comment by pipitone** [comment]
Gabe, can you have a look at this fix and let me know what you think.

**Comment by gdevenyi** [comment]
This fix up looks okay to me, other than to generalize the default class a bit further if possible.

**Comment by pipitone** [comment]
We could generalize but then there might be too much metaprogramming black magic. I'll have a think about it this evening.

**Comment by gdevenyi** [comment]
Lets handle this via #85

**Comment by gdevenyi** [comment]
Fixed by #240, merged into `dev`. When `-c` and `-j` are not given, they now default to `QBATCH_CHUNKSIZE` / `QBATCH_CORES`, and then to `--ppj`. So `--ppj 8` alone packs and runs 8 commands per job. It ships in 3.0.0.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01ERozY2aKk2XwaikUFTYRy9
